app/src/main/java/com/m/services/auth/LoginController.java

```diff
@@ -99,8 +99,12 @@ private void performTokenRequest(TokenRequest request) {
     private void receivedTokenResponse(
             @Nullable TokenResponse tokenResponse,
             @Nullable AuthorizationException authException) {
-        mAuthState.update(tokenResponse, authException);
-
-        listener.authComplete();
+            mAuthState.update(tokenResponse, authException);
+        if (authException == null) {
+            listener.authComplete();
+        } else {
+            System.out.println(authException.errorDescription);
+            reAuthorise();
```

Comment:
I am not sure if it's a good idea to just reauthorise when there the current auth process is failed. Would it better if we just pass the error message back to the user and let them decide what to do?

Reply:
After speaking with @MStokluska I have learned we did not need to check this error. App Auth package handles most of it and if it every does raise issue, the graphql server should give us a 401 or 403. These error we do handle.

Reply:
I think what happens is that when we are requesting a token a user is already past the authorization and we are only requesting a token. If we fail to receive a token for some reason ( ie network unavailable?) we won't be able to proceed with `get` query anyway and it will trigger 403 which is going to call refreshToken, so it kind of felt unnecessary to be checked in here as well but I might be wrong on this.

Reply:
ok, I think we can leave this as it is.